doc/source/user_guide/user_defined_functions.rst

```diff
@@ -0,0 +1,275 @@
+.. _user_defined_functions:
+
+{{ header }}
+
+**************************************
+Introduction to User-Defined Functions
```

Comment:
```suggestion
User-Defined Functions (UDFs)
```

This will be what will be shown in the index too, so better to be concise. Also, better for consistency to remove the `Introduction to`, which we could have in every other user guide too.